sky/sdk/lib/widgets/snack_bar.dart

```diff
@@ -43,47 +44,33 @@ class SnackBarAction extends Component {
   }
 }
 
-class SnackBar extends AnimatedComponent {
+// TODO(mpcomplete): generalize this to a SlideIn class.
+class SnackBarSlideInIntention extends AnimationIntention {
+  SnackBarSlideInIntention(this.duration, this.onStatusChanged);
 
-  SnackBar({
-    Key key,
-    this.content,
-    this.actions,
-    this.showing,
-    this.onStatusChanged
-  }) : super(key: key) {
-    assert(content != null);
-  }
-
-  Widget content;
-  List<SnackBarAction> actions;
-  bool showing;
+  Duration duration;
   SnackBarStatusChangedCallback onStatusChanged;
-
-  void syncFields(SnackBar source) {
-    content = source.content;
-    actions = source.actions;
-    onStatusChanged = source.onStatusChanged;
-    if (showing != source.showing) {
-      showing = source.showing;
-      showing ? _show() : _hide();
-    }
-  }
-
   AnimatedValue<Point> _position;
   AnimationPerformance _performance;
 
-  void initState() {
+  void initFields(AnimatedContainer container) {
     _position = new AnimatedValue<Point>(new Point(0.0, 50.0), end: Point.origin);
     _performance = new AnimationPerformance()
       ..duration = _kSlideInDuration
       ..variable = _position
-      ..addListener(_checkStatusChanged);
-    watch(_performance);
-    if (showing)
+      ..addListener(() { _updateProgress(container); });
+    _performance.progress = 0.0;
+    if (container.tag)
       _show();
   }
 
+  void syncFields(AnimatedContainer original, AnimatedContainer updated) {
+    if (original.tag != updated.tag) {
+      original.tag = updated.tag;
+      original.tag ? _show() : _hide();
+    }
```

Comment:
This seems unfortunate...  You could imagine wanting to mix in a bunch of intentions that had different state associated with them.  With this design, they'll stomp on the same tag field.  It's probably fine for this iteration, but something we might want to revisit as we use this system more.